README.md

```diff
@@ -1,3 +1,562 @@
+
+> [!WARNING]
+> The Findex server must be run with the following arguments:
+>
+> For example with Google OAuth2:
+>
+> ```sh
+> docker run -p 66688:6668 --name findex-server ghcr.io/cosmian/findex-server:0.1.0 \
+>            --https-p12-file=server/certificate.p12 \
+>            --https-p12-password=complex_password \
+>            --authority-cert-file=server/ca.crt
+> ```
+
+### User access authorization
+
+Once authenticated, users must have an authorization to an **Index ID**.
+
+#### What is an Index ID?
+
+An **Index ID** allows the partitioning of indexed data: behind an **Index ID**, there are:
+
+- encrypted indexes
+- and the encrypted version of the data that has been indexed.
+
+This **Index ID** can be shared between users, with a specific permission. Users can also own multiple Index IDs.
+
+For each server's endpoint, the server checks the user's permissions before allowing access to the index.
+
+#### Permissions
+
+```mermaid
+sequenceDiagram
+  autonumber
+  actor User
+  participant F as Findex server
+
+  User->>F: User authentication
+  User->>F: User create an Index ID
+  F->>User: Index ID with admin permission
+  User->>F: User grant/revoke permission to another user
+```
+
+There are 3 permissions:
+
+- `reader`
+- `writer`
+- `admin`
+
+The mechanism is pretty simple:
+
+When a user create a new **Index ID**, he becomes the **admin** of this index. He can then grant other users to the index with a role of **reader**, **writer** or **admin**.
+
+Every server endpoint is protected by this authorization mechanism: the server checks the user's role before allowing access to the endpoint.
+
+#### Endpoints
+
+| Endpoint                                              | Description                                       |
+| ----------------------------------------------------- | ------------------------------------------------- |
+| `/create/index`                                       | Create an **Index ID**                            |
+| `/permission/grant/{user_id}/{permission}/{index_id}` | Grant a permission to a user for a specific index |
+| `/permission/list/{user_id}`                          | List permissions of a user                        |
+| `/permission/revoke/{user_id}/{index_id}`             | Revoke a user's permission for a specific index   |
+
+### Encrypted database
+
+From the server's perspective, only encrypted data is received and stored as-is. The server does not know how the data is encrypted and cannot decrypt it.
+
+The requirements database is a key-value store where the keys are unique identifiers (UIDs) and the values are the encrypted indexes or datasets.
+
+In this scenario, the user is responsible for encrypting the data before sending it to the server. A hybrid encryption scheme is used, where the data is encrypted with a Data Encryption Key (DEK) and the DEK is encrypted with a Key Encryption Key (KEK).
```

Comment:
There is predefined scheme (in `cosmian_cli`) that the user is free to use. Otherwise, since `cosmian cli` gives him access to all the KMS encryption features, he can do it manually with its custom scheme

Reply:
Maybe make this clearer then.